### PR TITLE
New racial languages

### PR DESCRIPTION
**Pull request by EOBGames**
## About The Pull Request
Adds new languages for 8 of the alien species on the station- Ethereals, Plasmamen/Skeletons, Mothmen, Golems, Nightmares, Podpeople and Flypeople. Also adds a new tongue for ethereals, the electric discharger, giving ethereals a distinct crackle to their voice, as well as removing their sense of taste (because it's hard to taste with electricity).

## Why It's Good For The Game
Adds some much needed racial diversity to the species aboard the station, as well as contributing to the atmosphere of paranoia we all love so much.

## Changelog
:cl:
add: Following a recent incident involving the Rosetta Stone and an assistant, a flood of new languages have hit the station! Most roundstart races, as well as several special races, have been given their own languages to secretly chat with one another. Beware though, since the AI can understand the roundstart languages!
add: Voltaic, a new language for ethereals.
add: Moffic, a new language for moths.
add: Calcic, a chattery new language for plasmamen and skeletons. Bone golems get this one too.
add: Terrum, a new language for golems.
add: Shadowtongue, a new language for shadowlings and nightmares. (You n'wah!)
add: Sylvan, a new language for podpeople.
add: Buzzwords, a new "language" for flypeople.
add: the electric discharger, a new tongue for ethereals.
tweak: made a few changes to tongues for flies and skeletons.
tweak: Runic golems now get to speak Nar'sien.
imageadd: added icons for the new languages, and the electric discharger.
/:cl:

### CODE REVIEW

**Comment by Ohagi-Chan** [comment]
Any plans for a "racial tongue prefix", like how .h uses your main department?

**Comment by PKPenguin321** [comment]
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#banned-content

>Code adding, removing, or updating the availability of alien races/species/human mutants without prior approval. Pull requests attempting to add or remove features from said races/species/mutants require prior approval as well.

**Comment by kriskog** [comment]
I asked for this :)

**Comment by PKPenguin321** [comment]
Ok issue jannie

**Comment by kriskog** [comment]
Okay

**Comment by wesoda25** [comment]
Can we please not add unique languages to any more roundstart races? The golem, skeleton, shadowperson, and pod people languages seem fun and cool, but what does bilingual roundstart races add to the game? All that ever comes from it is one person only using their unique language every now and then, spamming comms with shit less than 10% of players understand.

**Comment by necromanceranne** [comment]
Because unique languages is fun and opens opportunity for keeping people out of the loop in a conversation you can have right in front of them.

Also, playing the weird foreigner that can't properly speak Galactic Common can be fun.

**Comment by Shadowflame909** [comment]
racial tension is wack man....

all space-men die when thrown out of an airlock.

**Comment by ATH1909** [comment]
Isn't access to their own special language what sets lizardpeople apart from the other roundstart races? This kind of dilutes that advantage, IMO.

Barely anyone uses draconic right now; these other racial languages seem like they're just bloat for the language list.

**Comment by ATH1909** [comment]
> racial tension is wack man....
> 
> all space-men die when thrown out of an airlock.
> 
> Edit: A more high IQ version of this is simply, kill-bait escalation /tg/ doesn't deserve features that allow non-antagonists to start more violence.
> 
> They're doing a whole lot with lizards and tables as is.

... what?

> all space-men die when thrown out of an airlock.

Golems don't.

> Edit: A more high IQ version of this is simply, kill-bait escalation /tg/ doesn't deserve features that allow non-antagonists to start more violence.

"Don't add more languages because people could kill/escalation-bait with them" makes about as much sense as "remove foam fingers because someone could kill/escalate-bait with them by hitting someone repeatedly with them for 0 damage"- in other words, it makes absolutely no fucking sense at all.

**Comment by carshalash** [comment]
No earth only language?

**Comment by Grazyn** [comment]
>Isn't access to their own special language what sets lizardpeople apart from the other roundstart races? This kind of dilutes that advantage, IMO.

Yeah that was the main reason given for closing all the previous PRs adding new languages to existent roundstart species.

**Comment by necromanceranne** [comment]
Draconic is hardly special if it is known by the silicons

**Comment by Grazyn** [comment]
>Draconic is hardly special if it is known by the silicons

It's still a unique gimmick for lizards. There are 3 main points that are taken into account when deciding whether a species should be roundstart:
- It must have one or more interesting and unique features so that it's not just a human reskin
- The feature shouldn't give them an advantage over the other species
- If it does give them an advantage, it should be balanced by one (or more) negative effects

If every other roundstart species gets its own language, lizards lose their unique gimmick, meaning they no longer fulfill the critera for being a roundstart species.

**Comment by Eskjjlj** [comment]
No felinid language???

**Comment by necromanceranne** [comment]
I do not genuinely see their language as being what sets lizards apart from other races. They are one of the only classical scifi elements of the game given most every other race is either a bad joke or is fantastical in some way. The other two are flypeople (a classic horror movie nod) which aren't round start, and plasmamen, which is ss13's original concept for a race...I think. And also a whole can of worms in of themselves given nobody can agree on what is right for them mechanically.

Lizardpeople have enough mechanical variance to inform their gameplay (obligate carnivores, heat resistant, cold vulnerable, optional vulnerability to the hardest stun in the game left behind because haha funne lizard dies to sharp glass, ignoring that someone added shoe snatching and made everyone vulnerable to glass shards so now the joke is on everyone who left a 20 second stun in the game that is easily achieved with little effort)

**Comment by JoshAdamPowell** [comment]
AI / borgs should be able to understand and speak all roundstart languages

**Comment by kriskog** [comment]
> 
> 
> AI / borgs should be able to understand and speak all roundstart languages

I believe they were supposed to.
@EOBGames needs to be added to the synthetic language holder. Consider if only roundstart races though.

**Comment by necromanceranne** [comment]
I'd recommend changing golemic into terran, after the Dungeons and Dragons language for earth elementals or creatures originating from the plane of earth. Mostly because ROCK BOIS. And also give Bone Golems Calcic.

On that same note, maybe make Shadwspeak 'Shadowtongue'.

**Comment by Ryll-Ryll** [comment]
But like when are you ever going to have more than one nightmare on station and when are they ever going to need to communicate in front of other people to each other without being overheard

**Comment by kriskog** [comment]
Well hopefully shadowpeople have it too? Those can be numerous in any given round.

**Comment by Shadowflame909** [comment]
> "Don't add more languages because people could kill/escalation-bait with them" makes about as much sense as "remove foam fingers because someone could kill/escalate-bait with them by hitting someone repeatedly with them for 0 damage"- in other words, it makes absolutely no fucking sense at all.

This would happen on highpop though.

High-Pop escalation was a mistake and they need all their sparks of conflict taken away.

Also golems are not spacemen!

**Comment by anconfuzedrock** [comment]
remember #34804?

**Comment by kriskog** [comment]
> 
> 
> remember #34804?

We already covered this. He has my permission.

**Comment by Timonkeyn** [comment]
Thanks, very creative language names, very cool

**Comment by Timonkeyn** [comment]
Where's catspeak for felinids

**Comment by ExcessiveUseOfCobblestone** [comment]
just make lizard's language more exclusive so it elevates itself into a noticeable benefit?

I'd also think it would be cool if there was a language that was quirk-bound so only people who have the quirk and librarian (and omnitongue) have access to it at the cost of whatever downsides you pick. 

Not terribly keen on giving EVERY race their own language, although it's more of a "detracting from the purpose of common radio" issue than a "human's first" one.

**Comment by Teepeepee** [comment]
I remember being told moth language (which was called lepidopterian, not moffic) were removed because they encouraged metagangs, how has that changed since?
also, if everyone gets sekrit-club speak, maybe humans could have english too, and other races only understand galactic common

**Comment by RaveRadbury** [comment]
Galactic Common is the human language. If you see someone speak it when you don't understand it you'll see that it's the Mandarin-English hybrid that futurists get all frothy over. It is called Galactic Common to demonstrate the power of humans as the cultural hegemony. They are the ones in charge, so their language is the "normal" language that everyone has to speak.

**Comment by kriskog** [comment]
Metagangs dont need a common race language to do their thing, they can just metagang normally. Besides, we allow and encourage racism ic so why would we discourage people of the same race sticking together?

As for the common channel, its full of spammy trash already. This won't make it any worse.

**Comment by Farquaar** [comment]
I don't really understand why flypeople and podpeople should have new languages. Languages are taught, not acquired through genetic mutation. It just doesn't fit with the lore/mechanics.

**Comment by PKPenguin321** [comment]
wait is kriskog an actual maintainer i thought he was a jannie

**Comment by carlarctg** [comment]
i was iffy on this but then i saw calcic, and then i saw the golem-hebrew thing, which is genius

**Comment by wesoda25** [comment]
> I don't really understand why flypeople and podpeople should have new languages. Languages are taught, not acquired through genetic mutation. It just doesn't fit with the lore/mechanics.

Pod People exist in isolation, it would make sense at least in that capacity.

**Comment by Farquaar** [comment]
> 
> 
> > I don't really understand why flypeople and podpeople should have new languages. Languages are taught, not acquired through genetic mutation. It just doesn't fit with the lore/mechanics.
> 
> Pod People exist in isolation, it would make sense at least in that capacity.

Maybe so, but most podpeople are made on the station via pod seeds. The terrarium life seeders are the exception, not the norm.

**Comment by 81Denton** [comment]
@PKPenguin321 kriskrog is a maintainer, if you see him on the servers please remind him that he's no longer allowed to play the game

**Comment by Mickyan** [comment]
We should totally make speaking racial languages a minor crime, but I have no idea what the process for changing space law is.

It would be an interesting way to show the cultural oppression of non-human races and at the very least dissuade people from using it openly over the radio without fear of retaliation.

**Comment by wesoda25** [comment]
> Maybe so, but most podpeople are made on the station via pod seeds. The terrarium life seeders are the exception, not the norm.

Maybe just give the language to them then

File: code/modules/language/language_holder.dm [hunk 1]
```diff
@@ -296,10 +296,74 @@ Key procs
...
+							/datum/language/terrum = list(LANGUAGE_ATOM),
+							/datum/language/calcic = list(LANGUAGE_ATOM))
+
+/datum/language_holder/runegolem
+	understood_languages = list(/datum/language/common = list(LANGUAGE_ATOM),
+								/datum/language/terrum = list(LANGUAGE_ATOM),
+								/datum/language/narsie = list(LANGUAGE_ATOM))
+	spoken_languages = list(/datum/language/common = list(LANGUAGE_ATOM),
+							/datum/language/terrum = list(LANGUAGE_ATOM),
+							/datum/language/narsie = list(LANGUAGE_ATOM))
```
Comment (kriskog) [inline code/modules/language/language_holder.dm]: These two should be children of the golem holder, ie /datum/language_holder/golem/bone.
You will still have to set all the languages, but it makes the structure a bit more obvious and allows for changing other stuff for all the golems if that ever becomes a thing.

**Review by kriskog — CHANGES_REQUESTED** [review]

File: code/modules/language/language_holder.dm (hunk 1 above)
Comment (EOBGames) [reply code/modules/language/language_holder.dm]: Issues sorted out.

**Review by kriskog — APPROVED** [review]

**Comment by MCterra10** [comment]
No exclusive human only language! This is discrimination!

**Pull request merged**